csrc/ir/nodes.cpp

```diff
@@ -4568,13 +4536,14 @@ std::vector<PolymorphicValue> MatmulOp::evaluate(
   const auto b = inputs.at(1).as<at::Tensor>();
 
   auto matmul_out = at::matmul(a, b);
-  if (ir_utils::hasTrivialAllocationDomain(out())) {
+  const auto& [sizes, strides] = inferShapeOfOutput(out(), ee);
+  auto options =
+      c10::TensorOptions().device(c10::Device(c10::DeviceType::Meta));
+  auto meta_out = at::empty_strided(sizes, strides, options);
```

Comment:
```suggestion
  auto meta_out = at::detail::empty_strided_meta(sizes, strides, a.dtype());
```
I haven't tested this, but some variations like this can probably save you some typing.